workflows/codefresh-csdp/versions/1.1.0/images/image-enricher-git-info/src/providers/bitbucket-server/bitbucket-server.service.js

```diff
@@ -0,0 +1,48 @@
+const _ = require('lodash');
+const bitBucketServerApi = require('./bitbucket-server.api');
+
+class BitbucketServerService {
+
+    async getPullRequestsWithCommits(repo, branch) {
+        try {
+            const prs = await bitBucketServerApi.getPullRequests(repo, branch);
+            return Promise.all(prs.map(async (pr) => {
+                const info = await bitBucketServerApi.getCommitsInfo(repo, pr);
+                const result = {
+                    ...info,
+                    prDate: pr.updatedDate || pr.createdDate,
+                    number: pr.id,
+                    url: `${pr.links.self[0].href}`,
+                    title: pr.title,
+                }
+
+                return result;
+            }));
+        } catch (error) {
+            throw new Error(`failed to get pull requests: ${error.message}`)
+        }
+    }
+
+    async getBranch(repo, branch) {
+        try {
+            const bitBucketBranch = await bitBucketServerApi.getBranch(repo, branch)
+            if (!bitBucketBranch) {
+                return null
+            }
+            return {
+                name: branch,
+                commit: bitBucketBranch.id,
+                commitMsg: bitBucketBranch.message,
+                commitURL: bitBucketBranch.commitURL,
+                committerUsername: _.get(bitBucketBranch, 'author.emailAddress')
+            }
+        } catch (error) {
+            if (_.get(error, 'response.statusCode') === 404) {
```

Comment:
I see that the same condition was used in gitlab.service. Just ensure that status code in bitbucket error response is stored in that path, because for example github has different error status code path.